doc/BuildPipeline.md

```diff
@@ -0,0 +1,50 @@
+The template for our mobile applications has been tweaked and refined over a long time. The latest version includes a fully functional YAML pipeline and uses [multiple stages](https://devblogs.microsoft.com/premier-developer/azure-devops-pipelines-multi-stage-pipelines-and-yaml-for-continuous-delivery/) in order to properly separate the different steps and reuse the code for similar ones. Here's a little rundown of the different files:
+
+# [.azure-pipelines.yml](../.azure-pipelines.yml)
+This is the point of entry for our builds. The sequence is as follows:
+  1. Build the application for the staging environment
+  2. Release the staging application to **AppCenter** _AND_ simultaneously build the application for the production environment
+  3. Release the production application to **AppCenter**, the **Google Play Store** and the **Apple AppStore** (the last two are usually configured to be run manually using [approvals](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/approvals?view=azure-devops&tabs=check-pass#approvals))
+
+## Pull request builds
+Due to the length of Xamarin build, this template is configured to behave a little differently when building a pull request (PR). We developed [a set of tools](https://github.com/nventive/MSBuild.UnifiedExtensions/blob/develop/src/Application.Building.Light/Readme.md) to reduce the build time at the cost of some optimizations. This requires a specific variable called `IsLightBuild` to be set, hence why it is appearing in the pipeline. 
+
+Also, being a PR build, we do not find it necessary to release it, so all the relevant stages are disabled.
+
+## Canary
+This template also includes build and release stages dedicated for Canary testing our packages. For the build side, the same templates as the other builds are used and we simply pass the required steps through variables. The release process is identical: we only release in AppCenter.
+
```

Comment:
What do you think of this? Is it necessary or it's redundant?
```suggestion
This step is only recommended for some projects at nventive, feel free to remove those from the yml if you don't need them but it won't impact anything if you leave the yml code.
```

Reply:
Thanks! I've rephrased a bit and added it!